src/main/java/org/owasp/esapi/Encoder.java

```diff
@@ -298,54 +332,55 @@
      *    &nbsp;&nbsp;window.setInterval('&lt;%= EVEN IF YOU ENCODE UNTRUSTED DATA YOU ARE XSSED HERE %&gt;');
      *  &lt;/script&gt;
      * </pre>
-     * @param input 
-     *          the text to encode for JavaScript
+     * @param untrustedData 
+     *          the untrusted data to output encode for JavaScript
      * 
-     * @return input encoded for use in JavaScript
+     * @return the untrusted data safely output encoded for use in a use in JavaScript
      */
-	String encodeForJavaScript(String input);
+    String encodeForJavaScript(String untrustedData);
 
-	/**
-	 * Encode data for insertion inside a data value in a Visual Basic script. Putting user data directly
-	 * inside a script is quite dangerous. Great care must be taken to prevent putting user data
-	 * directly into script code itself, as no amount of encoding will prevent attacks there.
-	 * 
-	 * This method is not recommended as VBScript is only supported by Internet Explorer
-	 * 
-	 * @param input 
-	 * 		the text to encode for VBScript
-	 * 
-	 * @return input encoded for use in VBScript
-	 */
-	String encodeForVBScript(String input);
+    /**
+     * Encode data for insertion inside a data value in a Visual Basic script. Putting user data directly
+     * inside a script is quite dangerous. Great care must be taken to prevent putting user data
+     * directly into script code itself, as no amount of encoding will prevent attacks there.
+     * 
+     * This method is not recommended as VBScript is only supported by Internet Explorer
+     * 
+     * @param untrustedData 
+     *      the untrusted data to output encode for VBScript
+     * 
+     * @return the untrusted data safely output encoded for use in a use in VBScript
+     */
+    String encodeForVBScript(String untrustedData);
```

Comment:
Nice switch.

Reply:
The discussion with the authors of the aforementioned arXiv paper where a commonly documented mistake was that participants were trying to use ESAPI Encoder's output encoding on the input (e.g., request parameters) rather than doing output encoding on the output side. I then noticed that most of the parameters were all called 'input' (which I switched to 'untrustedData' and then changed the @return for most of the HTTP related output encoders.